classes/logger.php

```diff
@@ -235,4 +235,17 @@ public static function get_existing_run_ids() {
         return $DB->get_fieldset_sql("SELECT runid FROM {" . self::TABLE . "} GROUP BY runid ORDER BY runid");
     }
 
+    /**
+     * Return last run for task
+     *
+     * @return array
+     */
+    public function get_last_execute_time() {
+        global $DB;
+
+        return $DB->get_field_sql(
+            "SELECT time FROM {" . self::TABLE . "} WHERE taskid = ? AND action = 'execute' AND info LIKE '%started%' ORDER BY time DESC LIMIT 1",
```

Comment:
this LIKE '%started%' seems a little brittle to me and potentially slower. Also shouldn't this be 'Task completed' rather than the start time?

Reply:
action = 'execute' is important. We could match it on 'Task started' if you wish. Started is better than completed because there may be records added between start and finish.

Reply:
action = 'execute' is fine, I wasn't suggesting changing it. The like smells to me, at minimum I think it should be exactly = 'Task started'.

>Started is better than completed because there may be records added between start and finish.

I don't understand this. If it fails or if it is still running then there might not be a complete, wouldn't we want the last full completion timestamp in both cases? Also any other records will be filtered out so I don't think they'd be in the brew no?

Reply:
'execute' logs are written regardless, to see if it is ok or not we need to see if there was an error logged in between, and even then we need to examine at which stage. Too heavy. This is the best we can do (I think) as a quick fix. Everything else is a project.